integrations/jina/src/haystack_integrations/components/embedders/jina/document_embedder.py

```diff
@@ -91,17 +97,25 @@ def to_dict(self) -> Dict[str, Any]:
         :returns:
             Dictionary with serialized data.
         """
-        return default_to_dict(
-            self,
-            api_key=self.api_key.to_dict(),
-            model=self.model_name,
-            prefix=self.prefix,
-            suffix=self.suffix,
-            batch_size=self.batch_size,
-            progress_bar=self.progress_bar,
-            meta_fields_to_embed=self.meta_fields_to_embed,
-            embedding_separator=self.embedding_separator,
-        )
+        kwargs = {
+            "api_key": self.api_key.to_dict(),
+            "model": self.model_name,
+            "prefix": self.prefix,
+            "suffix": self.suffix,
+            "batch_size": self.batch_size,
+            "progress_bar": self.progress_bar,
+            "meta_fields_to_embed": self.meta_fields_to_embed,
+            "embedding_separator": self.embedding_separator,
+        }
+        # Optional parameters, the following two are only supported by embeddings-v3 for now
+        if self.task:
+            kwargs["task"] = self.task
+        if self.dimensions:
+            kwargs["dimensions"] = self.dimensions
```

Comment:
If those are an empty string or `0` you'd miss the info.

```suggestion
        if self.task is not None:
            kwargs["task"] = self.task
        if self.dimensions is not None:
            kwargs["dimensions"] = self.dimensions
```